### PR TITLE
Initial version w/user-chooses semantics from w3c/mediacapture-main#667

### PR DESCRIPTION
**Pull request by jan-ivar**
Fixes https://github.com/w3c/mediacapture-extensions/issues/2 and https://github.com/w3c/mediacapture-extensions/issues/5 - Imported from https://github.com/w3c/mediacapture-main/pull/667 with intro and example added.

[Githack Preview](https://raw.githack.com/w3c/mediacapture-extensions/256e6adc160994d41a836c795fda0af2147caf23/index.html) (I tried and failed to get PR Preview to work on initial commit)

***

### :boom: Error: 500 Internal Server Error :boom: ###

[PR Preview](https://github.com/tobie/pr-preview#pr-preview) failed to build. _(Last tried on Nov 12, 2020, 4:35 PM UTC)_.

<details>
<summary>More</summary>

PR Preview relies on a number of web services to run. There seems to be an issue with the following one:

:rotating_light: [Spec Generator](https://www.w3.org/2015/labs/) - Spec Generator is the web service used to build specs that rely on ReSpec.

:link: [Related URL](https://labs.w3.org/spec-generator/?type=respec&url=https%3A%2F%2Frawcdn.githack.com%2Fw3c%2Fmediacapture-extensions%2F77e2fba08e231af2c486b26f3ab16ac6115ca0c9%2Findex.html%3FisPreview%3Dtrue)

```
🕵️‍♀️  That doesn't seem to be a ReSpec document. Please check manually: https://raw.githack.com/
```

_If you don't have enough information above to solve the error by yourself (or to understand to which web service the error is related to, if any), please [file an issue](https://github.com/tobie/pr-preview/issues/new?title=Error%20not%20surfaced%20properly&body=See%20w3c/mediacapture-extensions%239.)._
</details>

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -0,0 +1,353 @@
...
+      compared to the old semantics, they are somewhat incompatible with
+      expectations in some existing web applications that tend to call
+      {{MediaDevices/getUserMedia()}} repeatedly and lazily instead of using
+      e.g. <code>stream.clone()</code>.</p>
+    </section>
+    <section id="web compatibility">
+      <h3>Web compatibility and migration</h3>
+      <p>User agents are encouraged to provide the new semantics as opt-in
+      initially for web compatibility. User agents MUST deprecate (remove)
+      {{MediaDeviceInfo/label}} over time, though specific migration strategies
```
Comment (jan-ivar) [inline index.html:104]:
```suggestion
      {{MediaDeviceInfo/label}} from {{MediaDeviceInfo}} over time, though specific migration strategies
```

File: index.html [hunk 2]
```diff
@@ -0,0 +1,353 @@
...
+              SelectSettings</a>
+              algorithm, the value of <var>semanticsPresent</var>,
+              or any other internally-available information about
+              the devices, as inputs to its decision.
+              If <var>semantics</var> is <code>"user-chooses"</code>,
+              and the application has not narrowed down the choices
+              to one, then the user agent MUST ask the user to make
+              the final selection.</p>
+              <p>Once selected, the source of the
+              <code><a>MediaStreamTrack</a></code> MUST NOT change.</p>
```
Comment (henbos) [inline index.html:293]: This link is broken

Comment (dontcallmedom) [inline index.html:293]:
```suggestion
              {{MediaStreamTrack}} MUST NOT change.</p>
```

File: index.html [hunk 3]
```diff
@@ -0,0 +1,353 @@
...
+<head>
+  <title>Media Capture and Streams Extensions</title>
+  <meta content="text/html; charset=utf-8" http-equiv="Content-Type">
+  <script src="https://www.w3.org/Tools/respec/respec-w3c" class="remove"></script>
+  <script class='remove'>
+  "use strict";
+  // See https://github.com/w3c/respec/wiki/ for how to configure ReSpec
+  var respecConfig = {
+    group: "webrtc",
+    xref: ["html", "infra", "permissions", "dom", "mediacapture-streams"],
```
Comment (dontcallmedom) [inline index.html:12]:
```suggestion
    xref: ["html", "infra", "permissions", "dom", "mediacapture-streams", "webidl"],
```

**Review by henbos — APPROVED** [review]

File: index.html [hunk 4]
```diff
@@ -0,0 +1,353 @@
...
+          invoked.</p>
+        </li>
+        <li>
+          <p>Let <var>constraints</var> be the method's first argument.</p>
+        </li>
+        <li>
+          <p>Let <var>semanticsPresent</var> be <code>true</code> if
+          <var>constraints</var><code>.semantics</code> is
+          <a href="https://heycam.github.io/webidl/#dfn-present">present</a>,
+          otherwise <code>false</code>.</p>
```
Comment (dontcallmedom) [inline index.html:234]: this should use [= map/exists =] instead of "is present" (which WebIDL no longer defines)

File: index.html [hunk 5]
```diff
@@ -0,0 +1,353 @@
...
+  // See https://github.com/w3c/respec/wiki/ for how to configure ReSpec
+  var respecConfig = {
+    group: "webrtc",
+    xref: ["html", "infra", "permissions", "dom", "mediacapture-streams"],
+    edDraftURI: "https://w3c.github.io/mediacapture-extensions/",
+    editors:  [
+      {name: "Jan-Ivar Bruaroey", company: "Mozilla Corporation", w3cid: 79152},
+    ],
+    shortName: "mediacapture-extensions",
+    specStatus: "ED",
```
Comment (jan-ivar) [inline index.html:18]: From editor's meeting: Make this an unofficial draft for now
```suggestion
    specStatus: "UD",
```

File: index.html [hunk 6]
```diff
@@ -0,0 +1,353 @@
...
+          <p>Let <var>mediaDevices</var> be the object on which this method was
+          invoked.</p>
+        </li>
+        <li>
+          <p>Let <var>constraints</var> be the method's first argument.</p>
+        </li>
+        <li>
+          <p>Let <var>semanticsPresent</var> be <code>true</code> if
+          <var>constraints</var><code>.semantics</code> is
+          <a href="https://heycam.github.io/webidl/#dfn-present">present</a>,
```
Comment (jan-ivar) [inline index.html:233]:
```suggestion
          <var>constraints</var><code>.semantics</code> [= map/exists =],
```

File: index.html [hunk 7]
```diff
@@ -0,0 +1,352 @@
...
+  </section>
+  <section id="mediastreamconstraints-dictionary-extensions">
+    <h3>MediaStreamConstraints dictionary extensions</h3>
+    <div>
+      <pre class="idl"
+>partial dictionary MediaStreamConstraints {
+  GetUserMediaSemantics semantics;
+};</pre>
+      <section>
+        <h2>Dictionary <a class="idlType">MediaStreamConstraints</a>
```
Comment (jan-ivar) [inline index.html:157]:
```suggestion
        <h2>Dictionary {{MediaStreamConstraints}}
```

File: index.html [hunk 8]
```diff
@@ -0,0 +1,352 @@
...
+        Members</h2>
```
Comment (jan-ivar) [inline index.html:158]:
```suggestion
        <h2>Dictionary {{MediaStreamConstraints}}</h2>
```

File: index.html [hunk 9]
```diff
@@ -0,0 +1,351 @@
...
+  </section>
+  <section id="mediastreamconstraints-dictionary-extensions">
+    <h3>MediaStreamConstraints dictionary extensions</h3>
+    <div>
+      <pre class="idl"
+>partial dictionary MediaStreamConstraints {
+  GetUserMediaSemantics semantics;
+};</pre>
+      <section>
+        <h2>Dictionary {{MediaStreamConstraints}}</h2>
```
Comment (jan-ivar) [inline index.html:157]:
```suggestion
        <h2>Dictionary {{MediaStreamConstraints}} Members</h2>
```

**Pull request merged**